dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/split/orgunit/handler/DataOrgUnitSplitHandler.java

```diff
@@ -0,0 +1,83 @@
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.split.orgunit.handler;
+
+import javax.transaction.Transactional;
+
+import lombok.AllArgsConstructor;
+
+import org.hibernate.SessionFactory;
+import org.hisp.dhis.split.orgunit.OrgUnitSplitRequest;
+import org.springframework.stereotype.Service;
+
+/**
+ * @author Lars Helge Overland
+ */
+@Service
+@Transactional
+@AllArgsConstructor
+public class DataOrgUnitSplitHandler
+{
+    private static final String PARAM_ORG_UNIT = "organisationUnit";
+
+    private static final String PARAM_SOURCE = "source";
+
+    private final SessionFactory sessionFactory;
+
+    @Transactional
+    public void splitData( OrgUnitSplitRequest request )
+    {
+        migrate( request, "DataValueAudit", PARAM_ORG_UNIT );
+        migrate( request, "DataValue", PARAM_SOURCE );
+        migrate( request, "DataApprovalAudit", PARAM_ORG_UNIT );
+        migrate( request, "DataApproval", PARAM_ORG_UNIT );
+        migrate( request, "LockException", PARAM_ORG_UNIT );
+        migrate( request, "ValidationResult", PARAM_ORG_UNIT );
+        migrate( request, "MinMaxDataElement", PARAM_SOURCE );
+        migrate( request, "Interpretation", PARAM_ORG_UNIT );
+
+        migrate( request, "ProgramMessage", "recipients." + PARAM_ORG_UNIT );
+        migrate( request, "ProgramStageInstance", PARAM_ORG_UNIT );
+        migrate( request, "ProgramInstance", PARAM_ORG_UNIT );
+        migrate( request, "ProgramOwnershipHistory", PARAM_ORG_UNIT );
+        migrate( request, "TrackedEntityProgramOwner", PARAM_ORG_UNIT );
+        migrate( request, "TrackedEntityInstance", PARAM_ORG_UNIT );
+    }
+
+    private void migrate( OrgUnitSplitRequest request, String entity, String property )
+    {
+        String hql = String.format(
+            "update %s e set e.%s = :target where e.%s = :source",
+            entity, property, property );
+
+        sessionFactory.getCurrentSession().createQuery( hql )
+            .setParameter( "source", request.getSource() )
+            .setParameter( "target", request.getPrimaryTarget() )
+            .executeUpdate();
```

Comment:
would be nice if we could have debug logs here

Reply:
Okay added.